src/backend/src/main/kotlin/org/icpclive/overlay/Routing.kt

```diff
@@ -28,13 +29,52 @@ fun Route.configureOverlayRouting() {
     flowEndpoint("/mainScreen") { DataBus.mainScreenFlow.await() }
     flowEndpoint("/contestInfo") { DataBus.currentContestInfoFlow() }
     flowEndpoint("/runs") { DataBus.contestStateFlow.await().map { it.runsAfterEvent.values.sortedBy { it.time } } }
+    webSocket("/teamRuns") {
+        val teamIdStr = (incoming.receive() as? Frame.Text)?.readText()
+        if (teamIdStr.isNullOrBlank()) {
+            close(CloseReason(CloseReason.Codes.CANNOT_ACCEPT, "Invalid team id"))
+            return@webSocket
+        }
+        val teamId = teamIdStr.toTeamId()
+        sendJsonFlow(DataBus.contestStateFlow.await().map { state ->
+            state.runsAfterEvent.values.filter { it.teamId == teamId }
+                .sortedBy { it.time }
+        }.distinctUntilChanged()
+            .map { runs ->
+                val acceptedProblems = mutableSetOf<ProblemId>()
+                runs.map { info ->
+                    when (info.result) {
+                        is RunResult.ICPC -> {
+                            val icpcResult = info.result as RunResult.ICPC
+                            if (!acceptedProblems.contains(info.problemId)) {
+                                if (icpcResult.verdict == Verdict.Accepted) {
+                                    acceptedProblems.add(info.problemId)
+                                }
+                                TimeLineRunInfo.ICPC(info.time, info.problemId, icpcResult.verdict.isAccepted)
+                            } else {
+                                null
+                            }
+                        }
+
+                        is RunResult.IOI -> {
+                            val ioiResult = info.result as RunResult.IOI
+                            TimeLineRunInfo.IOI(info.time, info.problemId, ioiResult.scoreAfter)
```

Comment:
Not sure if we should use scoreAfter or score difference here. With score after, maybe we can show only improvements.